hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/mapreduce/HadoopPipelineMapper.java

```diff
@@ -0,0 +1,97 @@
+package org.molgenis.hadoop.pipeline.application.mapreduce;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.log4j.Logger;
+import org.molgenis.hadoop.pipeline.application.HadoopPipelineApplication;
+import org.molgenis.hadoop.pipeline.application.processes.BwaAligner;
+
+/**
+ * Hadoop MapReduce Job mapper.
+ */
+public class HadoopPipelineMapper extends Mapper<NullWritable, BytesWritable, NullWritable, Text> // SAMRecordWritable
+{
```

Comment:
Personal note for when the Reducer is also implemented. Currently `job.setNumReduceTasks(0)` in the `org.molgenis.hadoop.pipeline.application.HadoopPipelineApplication` makes sure no Reducer phase is done, though the last 2 fields ( `NullWritable, Text` ) are subject to change once a Reducer is implemented.

The `SAMRecordWritable` is a `Writable` from Hadoop-BAM (see https://github.com/HadoopGenomics/Hadoop-BAM and http://mvnrepository.com/artifact/org.seqdoop/hadoop-bam) that is able to handle `SAMRecord`from `htsjdk.samtools` (see https://github.com/samtools/htsjdk). A StreamReader converting the data to `SAMRecord` instances (instead of a single `String` instance in the current implementation) is currently in progress.